ballerina/vector_store.bal

```diff
@@ -183,21 +222,31 @@ public isolated class VectorStore {
 
     # Deletes vector entries from the store by their reference document ID.
     #
-    # + refDocId - The unique document ID used to identify and delete the corresponding vector entry.
-    # + return - An `ai:Error` if the deletion fails; otherwise, `()` is returned indicating success.
-    public isolated function delete(string refDocId) returns ai:Error? {
-        vector:DeleteRequest request = {
-            ids: [refDocId]
-        };
+    # + refDocIds - The unique document ID/IDs used to identify and delete the corresponding vector entries
+    # + return - An `ai:Error` if the deletion fails; otherwise, `()` is returned indicating success
+    public isolated function delete(string|string[] refDocIds) returns ai:Error? {
+        if refDocIds is string {
+            return self.deleteById(refDocIds);
+        }
+        foreach string id in refDocIds {
+            ai:Error? err = self.deleteById(id);
+            if err is ai:Error {
+                return err;
```

Comment:
This leads to an inconsistent stage.
Some are deleted and some are not.

Lets use the `batchAPI` like @MohamedSabthar mentioned or lets use a transactional block

Reply:
Fixed in [99c5b0d](https://github.com/ballerina-platform/module-ballerinax-ai.pinecone/pull/8/commits/99c5b0db3385b4c6c49df91d35e2a38903c3149f)